hpx/util/coroutine/coroutine.hpp

```diff
@@ -208,61 +205,14 @@ namespace hpx { namespace util { namespace coroutines
             std::move(target), id);
     }
 
-    //void reset()
-    //{
-    //    HPX_ASSERT(exited());
-    //    m_pimpl->reset();
-    //}
+    typedef typename arg_slot_traits::template at<0>::type arg0_type;
+    static const int arity = 1;
```

Comment:
What I am trying to get at is all that's involved here still looks overly complicated and bloated for a single known signature: `R(A)`, if I am not mistaken, it just looks like there is still a lot of potential for more simplification (removing most of the traits etc.)